statemachine/src/test/java/com/amazonaws/stepfunctions/cloudformation/statemachine/CreateHandlerTest.java

```diff
@@ -268,6 +283,86 @@ public void testDefinitionFromS3InYAML() throws Exception {
         assertThat(response.getResourceModel().getDefinitionString()).isEqualTo(formattedJson);
     }
 
```

Comment:
Can we add happy-path tests for each acceptable definition substitution type? 

E.g.
1. `testDefinitionSubstitutionWithStringTypes_shouldReturnDefinitionWithSubstitutionsMade`
2. `testDefinitionSubstitutionWithDoubleTypes_shouldReturnDefinitionWithSubstitutionsMade`
3. `testDefinitionSubstitutionWithBooleanTypes_shouldReturnDefinitionWithSubstitutionsMade`

Reply:
Can we also add unhappy-path tests for unacceptable definition substitution types?

E.g.

1. `testDefinitionSubstitutionWithUnacceptableType_{some unaccepted type}_shouldThrow{some exception}`

Reply:
+1, and can we separate them into dedicated test cases as Eliot mentioned above? That way we're not baking them into existing test cases which are meant to test other functionality.

Reply:
Since we’re not testing if the definition string after substitution is a valid state machine definition, but if the substitution is done correctly and if it is a valid json string, would it be necessary to divide them into dedicated test cases? Every line of the tests would be the same except for the substitution type, and I feel like this could be done in a single test.

Reply:
> Since we’re not testing if the definition string after substitution is a valid state machine definition

This is a weak point in our test coverage. These tests should assert that the actual output matches what's expected. I think we should begin doing that. Would you make that change in the next revision?

> would it be necessary to divide them into dedicated test case?

Ideally, a unit test has only one thing that can cause the test to fail. That's not always feasible, but in this case it's pretty easy to have. So, to answer your question, It's not exactly necessary but it would be an improvement.

Reply:
I added happy-path tests and unhappy-path tests as suggested